tests/test_modeling_utils.py

```diff
@@ -748,7 +748,10 @@ def test_load_state_dict(self):
 
             for param, device in device_map.items():
                 device = device if device != "disk" else "cpu"
-                assert loaded_state_dict[param].device == torch.device(device)
+                expected_device = (
+                    torch.device(f"{torch_device}:{device}") if isinstance(device, int) else torch.device(device)
+                )
+                assert loaded_state_dict[param].device == expected_device
```

Comment:
FYI this breaks CUDA tests because on CUDA we end up with `cuda:0:0`

Reply:
+1